Marlin/src/core/macros.h

```diff
@@ -244,7 +244,7 @@
 #define NEAR_ZERO(x) WITHIN(x, -0.000001f, 0.000001f)
 #define NEAR(x,y) NEAR_ZERO((x)-(y))
 
-#define RECIPROCAL(x) (NEAR_ZERO(x) ? 0 : (1 / float(x)))
+#define RECIPROCAL(x) ((x) ? 1.0f / float(x) : 0)
```

Comment:
For float values, this now returns a very large number for very small non-zero `x`, instead of 0. Is that behavior change intended?

Reply:
In general comparing a float or double with 0.0f is a bad idea. Because of the nature of floating point calculations, `for (float i=10.0; i==0.0; i-=10.0/3.0);` is likely to run forever. So comparing to a very small interval around 0 makes often sense.
Here we only want to avoid the division by zero. That would result in an error. And only the exact zero will result in the error. So here the test for the exact zero seems to be ok.

Reply:
Under this change `RECIPROCAL` is never used for variables, only for constants and `#define`s. For all other cases `1.0f / (value_known_to_be_good)` is currently employed. The reasoning being: to avoid having to do the comparisons when they are not needed.

Reply:
So, I guess I just made the case to restore it to its former "more safe" version.